// node-karin/dist/root.mjs
import path from 'path';
import { fileURLToPath } from 'url';

// src/root.ts
var filename = fileURLToPath(import.meta.url);
var karinPathRoot = Object.freeze(path.join(filename, "../..").replace(/\\/g, "/"));
var karinPathMain = Object.freeze(path.join(karinPathRoot, "dist", "index.js"));
var karinPathPlugins = Object.freeze(path.join(process.cwd(), "plugins"));
var isPackaged = Object.freeze(filename.includes("node_modules"));
var karinPathDefaultConfig = Object.freeze(path.join(karinPathRoot, "default", "config"));
var karinPathDefaultView = Object.freeze(path.join(karinPathRoot, "default", "view"));
var karinPathComment = Object.freeze(path.join(karinPathRoot, "default", "comment"));
var karinPathBase = Object.freeze(path.join(process.cwd(), "@karinjs"));
var karinPathConfig = Object.freeze(path.join(karinPathBase, "config"));
var karinPathData = Object.freeze(path.join(karinPathBase, "data"));
var karinPathTemp = Object.freeze(path.join(karinPathBase, "temp"));
var karinPathResource = Object.freeze(path.join(karinPathBase, "resource"));
var karinPathDb = Object.freeze(path.join(karinPathData, "db"));
var karinPathTaskDb = Object.freeze(path.join(karinPathDb, "task"));
var karinPathRedisSqlite3 = Object.freeze(path.join(karinPathDb, "redis-sqlite3"));
var karinPathKv = Object.freeze(path.join(karinPathDb, "kv"));
var karinPathLogs = Object.freeze(path.join(karinPathBase, "logs"));
var karinPathHtml = Object.freeze(path.join(karinPathTemp, "html"));
var karinPathPm2Config = Object.freeze(path.join(karinPathConfig, "pm2.yaml"));
var karinPathConsole = Object.freeze(path.join(karinPathTemp, "console"));
var karinPathSandboxData = Object.freeze(path.join(karinPathData, "sandbox"));
var karinPathSandboxTemp = Object.freeze(path.join(karinPathTemp, "sandbox"));
var karinDir = karinPathRoot;
var karinMain = karinPathMain;
var pluginDir = karinPathPlugins;
var isPkg = isPackaged;
var defaultConfigPath = karinPathDefaultConfig;
var defaultViewPath = karinPathDefaultView;
var commentPath = karinPathComment;
var basePath = karinPathBase;
var configPath = karinPathConfig;
var dataPath = karinPathData;
var tempPath = karinPathTemp;
var resourcePath = karinPathResource;
var dbPath = karinPathDb;
var redisSqlite3Path = karinPathRedisSqlite3;
var kvPath = karinPathKv;
var logsPath = karinPathLogs;
var htmlPath = karinPathHtml;
var pm2Path = karinPathPm2Config;
var consolePath = karinPathConsole;
var sandboxDataPath = karinPathSandboxData;
var sandboxTempPath = karinPathSandboxTemp;
var root_default = {
  // 新常量
  karinPathRoot,
  karinPathMain,
  karinPathPlugins,
  isPackaged,
  karinPathDefaultConfig,
  karinPathDefaultView,
  karinPathComment,
  karinPathBase,
  karinPathConfig,
  karinPathData,
  karinPathTemp,
  karinPathResource,
  karinPathDb,
  karinPathRedisSqlite3,
  karinPathKv,
  karinPathLogs,
  karinPathHtml,
  karinPathPm2Config,
  karinPathConsole,
  karinPathSandboxData,
  karinPathSandboxTemp,
  karinPathTaskDb,
  // 旧常量，保持兼容性
  karinDir,
  karinMain,
  pluginDir,
  isPkg,
  defaultConfigPath,
  defaultViewPath,
  commentPath,
  basePath,
  configPath,
  dataPath,
  tempPath,
  resourcePath,
  dbPath,
  redisSqlite3Path,
  kvPath,
  logsPath,
  htmlPath,
  pm2Path,
  consolePath,
  sandboxDataPath,
  sandboxTempPath
};

export { basePath, commentPath, configPath, consolePath, dataPath, dbPath, root_default as default, defaultConfigPath, defaultViewPath, htmlPath, isPackaged, isPkg, karinDir, karinMain, karinPathBase, karinPathComment, karinPathConfig, karinPathConsole, karinPathData, karinPathDb, karinPathDefaultConfig, karinPathDefaultView, karinPathHtml, karinPathKv, karinPathLogs, karinPathMain, karinPathPlugins, karinPathPm2Config, karinPathRedisSqlite3, karinPathResource, karinPathRoot, karinPathSandboxData, karinPathSandboxTemp, karinPathTaskDb, karinPathTemp, kvPath, logsPath, pluginDir, pm2Path, redisSqlite3Path, resourcePath, sandboxDataPath, sandboxTempPath, tempPath };
